modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -179,16 +195,13 @@ public TxState state(UUID txId) {
 
     @Override
     public void changeState(UUID txId, TxState before, TxState after) {
-        TxState computeResult = states.compute(txId, (k, v) -> {
+        states.compute(txId, (k, v) -> {
             if (v == before) {
                 return after;
             } else {
                 return v;
             }
         });
-
-        assert computeResult == after : "Unable to change transaction state, expected = [" + before + "],"
```

Comment:
I'm not sure if it is right to delete this asset because the redundant state changes invoked are not expected.

Reply:
Agree, assert seems useful.

Reply:
This assertion is no longer in the master.